02.calendar/calendar.rb

```diff
@@ -0,0 +1,84 @@
+require 'date'
+require 'optparse'
+
+class Calendar
+  # 年月の先頭のスペースの数
+  HEADER_MARGIN_SPACES = 6
+  # 1日分の半角スペースの数
+  ONE_DAY_SPACES = 3
+  MONTH_LABEL = "月"
+  DAYS = "日 月 火 水 木 金 土"
+  CHARACTER_COLOR_BLACK = "\e[30m"
+  BACKGROUND_COLOR_WHITE = "\e[47m"
+  RESET_CODE = "\e[0m"
+
+  def generate
+    year, month, errors = initialize_year_month
+    return errors.join("\n") unless errors.empty?
+
+    # 年月の先頭の余白
+    header_spaces = "\s" * HEADER_MARGIN_SPACES
+    # 第1週の先頭の余白
+    first_week_spaces = "\s" * ONE_DAY_SPACES * Date.new(year, month, 1).wday
+    # 表示月の最終日
+    last_day = Date.new(year, month, -1).day
+    # 表示する日付
+    dates = ""
+    (1..last_day).each do |day|
+      current_date = Date.new(year, month, day)
+      dates << "#{CHARACTER_COLOR_BLACK}#{BACKGROUND_COLOR_WHITE}" if current_date == today
+      dates << sprintf("%2d", day)
+      dates << RESET_CODE if current_date == today
+      # 最終日は空白も改行も不要
+      unless day == last_day
+        if current_date.saturday?
+          dates << "\n"
+        else
+          dates << "\s"
+        end
+      end
+    end
+    <<~EOF
+    #{header_spaces}#{month}#{MONTH_LABEL}\s#{year}
+    #{DAYS}
```

Comment:
```suggestion
          #{month}月 #{year}
    日 月 火 水 木 金 土
```

好みの問題かもしれませんが、明確なメリットがない場合は定数や変数をあまり多用しない方がシンプルでスッキリしたコードになるかもしれません。

Reply:
指摘していただいて、確かに変数や定数を多用することで却って読みづらくなってしまっていることに気づきました。
提案いただいた内容に修正しました。